impeller/renderer/backend/vulkan/texture_vk.h

```diff
@@ -7,29 +7,52 @@
 #include "flutter/fml/macros.h"
 #include "impeller/base/backend_cast.h"
 #include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "impeller/renderer/backend/vulkan/swapchain_vk.h"
 #include "impeller/renderer/backend/vulkan/vk.h"
 #include "impeller/renderer/texture.h"
 
 namespace impeller {
 
+enum class TextureBackingTypeVK {
+  kAllocatedTexture,
+  kWrappedTexture,
+};
+
+struct WrappedTextureInfoVK {
+  SwapchainImageVK* swapchain_image = nullptr;
+};
+
+struct AllocatedTextureInfoVK {
+  VmaAllocator* allocator = nullptr;
+  VmaAllocation allocation = nullptr;
+  VmaAllocationInfo allocation_info = {};
+  VkImage image = nullptr;
+};
+
+struct TextureInfoVK {
+  TextureBackingTypeVK backing_type;
+  union {
```

Comment:
std::variant instead?

Reply:
👍🏽  will move to wrapped texture as a new subclass in a follow-up PR.